### PR TITLE
Tightening the board states

### PR DESCRIPTION
**Pull request by xavierlepretre**
Adding some tests too.

### CODE REVIEW

File: contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt [hunk 1]
```diff
@@ -133,82 +145,102 @@ data class PlacedHexTiles(val value: MutableList<HexTile>) {
...
-                   val hexTileIndex: HexTileIndex,
-                   val sides: HexTileNeighbors = HexTileNeighbors(),
-                   var roads: MutableList<UniqueIdentifier?> = MutableList(SIDE_COUNT) { null }) {
-
-    init {
-        require(roads.size == SIDE_COUNT) { "roads.size cannot be ${roads.size}" }
-    }
+data class HexTile(
+        val resourceType: HexTileType,
+        val roleTrigger: Int,
```
Comment (xavierlepretre) [inline contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt]: @rogersanick 

* Should this be `rollTrigger`?
* I thought first that this is a number between `2` and `12`, but I have seen you set it to `0`? What is this?

Comment (rogersanick) [reply contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt]: Hey @xavierlepretre - the rollTrigger is set to 0 where the hexTile is of a type that should never be triggered (e.g. 'Desert').

We could also change the type to be `RollTrigger?` but that might break the filtering logic in issue resources flow where the initiating player must sort for valid settlements.

File: oracleClient/src/main/kotlin/com/oracleClientStatesAndContracts/states/DiceRollState.kt [hunk 2]
```diff
@@ -28,4 +18,49 @@ data class DiceRollState(
...
+    init {
+        require(real || total == 0) {
+            "Not real one should have a total of 0"
+        }
+        require(!real || 2 * DiceRollState.MIN_ROLL <= total) {
+            "A real one's total has to larger than ${DiceRollState.MIN_ROLL * 2}"
+        }
+        require(!real || total <= 2 * DiceRollState.MAX_ROLL) {
+            "A real one's total has to smaller than ${DiceRollState.MAX_ROLL * 2}"
+        }
```
Comment (xavierlepretre) [inline oracleClient/src/main/kotlin/com/oracleClientStatesAndContracts/states/DiceRollState.kt:64]: @rogersanick I created this `RollTrigger`, that has 2 situations. Wonder what is better. This or using a nullable or using an interface with 2 implementations.

**Comment by xavierlepretre** [comment]
@rogersanick I think it is a good moment to merge it. Lotsa work in.

**Comment by rogersanick** [comment]
Thanks @xavierlepretre - your contributions are hugely appreciated! Looking forward to Corda Code Club tomorrow.

**Pull request merged**